enhancements/network/on-prem-service-load-balancers.md

```diff
@@ -0,0 +1,370 @@
+---
+title: on-prem-service-load-balancers
+authors:
+  - "@russellb"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-05-14
+last-updated: 2020-05-14
+status: implementable
+---
+
+# Service Load Balancers for On Premise Infrastructure
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+We do not currently support [Services of
+type=LoadBalancer](https://kubernetes.io/docs/concepts/services-networking/#loadbalancer)
+(Service Load Balancers, or SLBs) for OpenShift in bare metal environments.
+While bare metal clusters are of primary interest, we hope to find a solution
+that would apply to other on-premise environments that don’t have native load
+balancer capabilities available (a cluster on VMware, RHV, or OpenStack, for
+example).
+
+## Motivation
+
+Service Load Balancers are a common way to expose applications on a cluster. We
+highly value clusters using on premise infrastructure, but do not support this
+feature in that context.  We aim to fill this gap with an optional range of
+capabilities.
+
+### Goals
+
+A SLB solution must provide these high level features:
+
+* Management of one or more pools of IP addresses to be allocated for SLBs
+* High Availability (HA) management of these IP addresses once allocated.  We
+  must be able to fail over addresses in less than 5 seconds.
+* Advertisement of the location of an IP address once assigned to a Node or
+  moved to a new Node.  Possible approaches include:
+    * Layer 2 (gratuitous ARP for IPv4, NDP for IPv6) - good for wide range of
+      environment compatibility, but limiting for larger clusters
+    * Layer 3 (BGP) - good for integration with networks for larger clusters
+* Suitable for large scale clusters (target up to 2000 nodes).
+* Single Node IPs (all traffic for a Service LB going through a single Node) is
+  acceptable for our out-of-the-box solution.  Support for a multi-Node based
+  solution would be nice, but is not required.
+* Must be compatible with at least the following cluster network types:
+  [OpenShift-SDN](https://github.com/openshift/sdn) and
+  [OVN-Kubernetes](https://github.com/openshift/ovn-kubernetes)
```

Comment:
use the upstream link. https://github.com/ovn-org/ovn-kubernetes
(we don't want a solution that depends on patches in our fork)

Reply:
will do, thanks